config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/GenericEntityBuilder.java

```diff
@@ -0,0 +1,89 @@
+import java.util.stream.Collectors;
+
+import static com.ca.apim.gateway.cagatewayconfig.bundle.builder.EntityBuilderHelper.getEntityWithOnlyMapping;
+import static com.ca.apim.gateway.cagatewayconfig.util.gateway.BundleElementNames.*;
+import static com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentUtils.createElementWithAttributesAndChildren;
+import static com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentUtils.createElementWithTextContent;
+
+public class GenericEntityBuilder implements EntityBuilder {
+
+    private static final Integer ORDER = 1700;
+    private final IdGenerator idGenerator;
+    private final DocumentTools documentTools;
+
+    @Inject
+    public GenericEntityBuilder(IdGenerator idGenerator, DocumentTools documentTools) {
+        this.idGenerator = idGenerator;
+        this.documentTools = documentTools;
+    }
+
+    @Override
+    public List<Entity> build(Bundle bundle, BundleType bundleType, Document document) {
+        Map<String, GenericEntity> genericEntities =
+                Optional.ofNullable(bundle.getGenericEntities()).orElse(Collections.emptyMap());
+        return buildEntities(genericEntities, bundle, bundleType, document);
+    }
+
+    private List<Entity> buildEntities(Map<String, GenericEntity> genericEntities, Bundle bundle,
+                                       BundleType bundleType, Document document) {
+        switch (bundleType) {
+            case DEPLOYMENT:
+                return genericEntities.entrySet().stream()
+                        .map(entry -> getEntityWithOnlyMapping(EntityTypes.GENERIC_TYPE, entry.getKey(),
+                                generateId(entry.getValue())))
+                        .collect(Collectors.toList());
+            case ENVIRONMENT:
+                return genericEntities.entrySet().stream()
+                        .map(e -> buildGenericEntity(e.getKey(), e.getValue(), document))
+                        .collect(Collectors.toList());
+            default:
+                throw new EntityBuilderException("Unknown bundle type: " + bundleType);
+        }
+    }
+
+    private Entity buildGenericEntity(String name, GenericEntity genericEntity, Document document) {
+        String id = generateId(genericEntity);
+        Element genericEntityElement = createElementWithAttributesAndChildren(
+                document,
+                GENERIC_ENTITY,
+                ImmutableMap.of(ATTRIBUTE_ID, id),
+                createElementWithTextContent(document, NAME, name),
+                createElementWithTextContent(document, DESCRIPTION, genericEntity.getDescription()),
+                createElementWithTextContent(document, ENTITY_CLASS_NAME, genericEntity.getEntityClassName()),
+                // Enabled element must be after Description and EntityClassName
+                createElementWithTextContent(document, ENABLED, Boolean.TRUE.toString()),
+                createElementWithTextContent(document, VALUE_XML, genericEntity.getValueXml())
```

Comment:
We have to remove id from xml in GenericEntityLinker and update the xml in this builder

<l7:ValueXml>&lt;?xml version="1.0" encoding="UTF-8"?&gt;
&lt;java version="1.8.0_242" class="java.beans.XMLDecoder"&gt;
 &lt;object class="com.l7tech.external.assertions.http2.routing.model.Http2ClientConfigurationEntity"&gt;
  &lt;void property="description"&gt;
   &lt;string&gt;&lt;/string&gt;
  &lt;/void&gt;
  &lt;void property="id"&gt;
   &lt;string&gt;b9020f62133b1e4d09c087abcbca2666&lt;/string&gt;
  &lt;/void&gt;
  &lt;void property="name"&gt;
   &lt;string&gt;http2config&lt;/string&gt;
  &lt;/void&gt;
  &lt;void property="valueXml"&gt;
   &lt;string&gt;&lt;/string&gt;
  &lt;/void&gt;
  &lt;void property="version"&gt;
   &lt;int&gt;1&lt;/int&gt;
  &lt;/void&gt;
 &lt;/object&gt;
&lt;/java&gt;
</l7:ValueXml>

Reply:
@uppoju We are not modifying the ValueXml and I don't think this Id has any effect on the object that gets created so not required I suppose. I tested the deployment with the same ID as in export and by modifying the ID in `@bundle-entity` annotation. Both worked!

Reply:
OK